### ISSUE TITLE
Lumen - Route name suggestion is missing automatically applied dots in names

### ISSUE THREAD

**Issue by sykorax** [laravel-idea/plugin#247]
Hello there,

First of all, I'm a huge fan of what you're doing for us!

I have a project written in Lumen. I just realised how Lumen concatenates route names by itself.
Now, if we have a group which has an 'as' attribute, then named sub routes will inherit the name with a dot (.) infix automatically.

```php
$router->group(['as' => 'foo'], function () use ($router) {

    $router->get('', ['as' => 'bar']);
});
```
This will result the route name as: ``foo.bar``, please note that inserting dot is not required anymore.

Laravel IDEA offers the route as:
![image](https://user-images.githubusercontent.com/5334912/111067879-8c780900-84c6-11eb-866c-1d49660ada62.png)

so it's basically missing the dot.
I am not sure if the same behavior applies to Laravel as well.

Thank you for your work!

**Comment by adelf** [comment]
Hi, Forgács. Thanks for sharing this. Now I have to find which versions of Laravel and Lumen have this behavior and which ones are different :)

**Comment by sykorax** [comment]
Hi, thanks for the quick response!
I did some research.
It seems it has always been like this.

As i can see it was included in the very [early version](https://github.com/laravel/lumen-framework/commit/7018d57d684e7e63bcef9576bc718e9ad75feb0c#diff-cc75cf1c7e7edc826fe7fbc1a2d190a5e8628666c5108e2566945c90e85b03adR100) of Lumen (5.X)

So maybe it's a Lumen specific stuff and it has nothing to do to Laravel at all since they differ in routing from both API and implementation perspective. Now i'm using 8.X for my project and it works exactly the same way. Maybe you can apply this name resolver change on Lumen frameworks regardless of its version. What do you think?

**Comment by adelf** [comment]
I’ll check tomorrow. If it’s just a Laravel/Lumen difference - that will be very easy.

**Comment by adelf** [comment]
Fixed in 4.2.1
